### ISSUE TITLE
Some windows forever lost if they have  _NET_WM_STATE_SKIP_TASKBAR property

### ISSUE THREAD

**Issue by astier** [sagb/alttab#147]
### Problem

I am using a custom WM which maximizes all windows, without a taskbar. My only means of changing windows is using alttab. Unfortunately some dialog-windows like the setting-window from pycharm (a jetbrains ide) has the property `_NET_WM_STATE_SKIP_TASKBAR` set. This means if I ever change the window from the setting-window to another window I never can focus the setting-window again because it is hidden behind the other windows and alttab ignores it because of the `_NET_WM_STATE_SKIP_TASKBAR` property. As a side-effect pycharm becomes totally unusable because it doesn't receive any key-/button-presses as long as the setting-window is open. I have to focus the setting-window manually from the terminal with `wmctrl -a Setting` and close it to make pycharm responsive again. For some reason if I use `herbstluftwm` alttab doesn't have this issue.

### Expected behavior

I want to be able to list and focus windows with alttab which also have the _NET_WM_STATE_SKIP_TASKBAR-property set. I am not sure what the best way to fix this would be. Should alttab stop ignoring _NET_WM_STATE_SKIP_TASKBAR-windows? Introduce an option to not ignore them? Or should it be handled on the wm-side (I am not sure how herbstluftwm fixes it)?

**Comment by astier** [comment]
Could it have anything to do with the fact that herbstluftwm is reparenting and my wm is not reparenting?

**Comment by astier** [comment]
I just checked and if I open the pycharm-setting-window in herbstluftwm it doesn't have _NET_WM_STATE_SKIP_TASKBAR set. Maybe it gets deleted by hlwm.

**Comment by sagb** [comment]
Setting/unsetting _NET_WM_STATE_SKIP_TASKBAR is likely pycharm's own decision.

I've used WMs which maximize all windows, without a taskbar, for years. Sometimes modal windows have not been recognized by alttab, probably for the reason you described here, but rarely and apps didn't appear unusable, so it was never be a problem for me.
Nevertheless, I'm going to add an option for this.

**Comment by sagb** [comment]
Please try `alttab -ns`.

**Comment by astier** [comment]
It works. Thank you. With this option enabled I can let the wm decide if a window should be ignored by alttab or not by (not) adding the respective window to `_NET_CLIENT_LIST_STACKING`. You can close the issue if you want.

**Comment by astier** [comment]
But if you want I also can just configure my wm to delete the _NET_WM_STATE_SKIP_TASKBAR property from windows which I want to show up in alttab if you don't want to deal with the additional code and think this would be a more idiomatic solution. Whatever you prefer I am fine with both.

**Comment by sagb** [comment]
Let it be altab feature.

**Comment by astier** [comment]
> I've used WMs which maximize all windows, without a taskbar, for years.

May I ask you what you are using now?

**Comment by sagb** [comment]
Ratpoison. Probably should have been switched to stumpwm, but they have obfuscated keybinding stuff too hard.
Herbstluftwm is probably also worth trying.
What is your custom WM?

**Comment by astier** [comment]
Just something I wrote myself. It's a private repo. The only thing it does is maximizing all windows, managing a couple ewmh-atoms so other software can work nicely with it (like alttab) and thats it basically. I use alttab to change windows and sxhkd to manage keybindings. ~300SLOC But I think I will have to add soon the ability to see two windows beside each-other. I started out with dwm but eventually removed almost all code which I didn't need and rewrote the rest.

**Comment by astier** [comment]
Eventually I realized that I don't need different desktops and tiling. Just maximizing and stacking is enough since I use tmux to manage my terminal-windows. Alttab was the last piece I needed to be able to switch efficiently between windows. Thanks for the great work! But as I said eventually I will have to tile one window on the left and one the right for my work. But it still will be stacking also. I will be able to stack windows on the left side and also to stack windows on the right side but also maximize them. Basically a tiling-stacking wm. I call it split-and-stack-wm.
